consensus/src/dag/types.rs

```diff
@@ -281,6 +332,8 @@ impl Deref for CertifiedNode {
 
 impl TDAGMessage for CertifiedNode {
     fn verify(&self, verifier: &ValidatorVerifier) -> anyhow::Result<()> {
+        ensure!(self.digest() == self.calculate_digest(), "invalid digest");
+
```

Comment:
Same comment. If we already have this certified node, why compute digests?

Reply:
We need digest to lookup the certifiednode.

Reply:
I think Zekun is changing it. Lookup by metadata.